### PR TITLE
Basic chaining support

### LINKED ISSUE #109: Chaining wrong
Indentation is wrong when chaining methods and using callables.

Before/after:
```diff
     public function jsonSerialize()
     {
-        return collect(parent::jsonSerialize())->mapWithKeys(function ($value, $key) {
-            return [camel_case($key) => $value];
-        })->toArray();
+        return collect(parent::jsonSerialize())->mapWithKeys(
+            function ($value, $key) {
+                return [camel_case($key) => $value];
+            }
+        )->
+            toArray(
+
+        );
     }
```

Thanks for your work!

### LINKED ISSUE #116: Reformating chaining
Same here, I can understand why it made that, but I think this is not a good idea:

```diff
-        Route::prefix('api')
-             ->middleware('api')
-             ->namespace($this->namespace)
-             ->group(base_path('routes/api.php'));
+        Route::prefix('api')->middleware('api')->namespace($this->namespace)->group(
+            base_path('routes/api.php')
+        );
```

### PR DESCRIPTION
**Pull request by czosel**
This is basically a simplified copy of what is done in the JS printer (see [here](https://github.com/prettier/prettier/blob/2e59f178aa174b35989d6e1a168032a48626dfc6/src/language-js/printer-estree.js#L3886))

Fixes #109 
Fixes #116

### CODE REVIEW

**Comment by evilebottnawi** [comment]
@czosel awesome work :+1:

**Comment by czosel** [comment]
@evilebottnawi shall we merge this? :smile:

File: src/printer.js [hunk 1]
```diff
@@ -130,6 +130,107 @@ function printLines(path, options, print) {
...
+  //       )
+  //     ))
+  //   ))
+  // and we transform it into (notice the reversed order)
+  //   [Identifier a, Call, PropertyLookup b, PropertyLookup c, Call,
+  //    PropertyLookup d, Call]
+  const printedNodes = [];
+
+  // recursive call to traverse AST
+  function rec(path) {
```
Comment (mgrip) [inline src/printer.js]: maybe give this a more semantic name?

Comment (czosel) [reply src/printer.js]: i thought the same when i copied it over from the JS printer, and added the comment instead :smile: I'll rename it and remove the comment. How about `traverse`?

Comment (mgrip) [reply src/printer.js]: sounds great!

**Pull request merged**